core/graphelier-service/db/mongo.go

```diff
@@ -336,52 +337,110 @@ func (c *Connector) GetTopOfBookByInterval(instrument string, startTimestamp uin
 	}
 	log.Tracef("Documents in interval {%d,%d}=%d, keeping 1 in %d", startTimestamp/interval, endTimestamp/interval, count, count/maxCount)
 
-	// Add $mod comparator to interval_multiple filter
-	filter[1].Value = append(filter[1].Value.(bson.D), bson.E{Key: "$mod", Value: bson.A{count / maxCount, 0}})
+	if maxCount < count {
```

Comment:
The function is getting pretty massive, can it be split up?

Reply:
it's responsible for the large and small intervals depending on the number of data points we receive from the frontend. if you want we can pass the cursor into another function to just get the results for the large interval?

Reply:
Yeah that works